components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/factory/DCRMgtOGSiServiceFactory.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.auth.valve.factory;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.oauth.dcr.DCRConfigurationMgtService;
+
+/**
+ * Since this factory produces DCRConfigurationMgtService connector service,  there is a possibility that said
+ * connector not available in the distribution.
+ * So rather than designing as Factory Beans this class designed as Singleton.
+ */
+public class DCRMgtOGSiServiceFactory {
+
+    private static DCRConfigurationMgtService dcrConfigurationMgtService = null;
+    private static final Log log = LogFactory.getLog(DCRMgtOGSiServiceFactory.class);
+
+
+    /**
+     * This method return  the instance if the OSGi service exists.
+     * Else throw Null pointer Exception. We handle the exception gracefully.
+     *
+     * @return DCRConfigurationMgtService
+     */
+    public static DCRConfigurationMgtService getInstance() {
+
+        if (dcrConfigurationMgtService == null) {
+            /* Try catch statement is included due to a  NullPointerException which occurs at the server startup and
```

Comment:
without catching the NPE,isn't there a way to check for null b4 exception appears

Reply:
Followed the existing similar approach as [here](https://github.com/wso2/identity-api-server/blob/1ebe8c5568926d8a453160fb571d5d440d5eb2d9/components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/factory/JWTAuthenticationMgtOGSiServiceFactory.java#L62)